mp/src/game/shared/momentum/weapon/weapon_mom_stickybomblauncher.cpp

```diff
@@ -0,0 +1,436 @@
+        m_flChargeBeginTime = gpGlobals->curtime;
+
+        if (m_bIsChargeEnabled.Get())
+        {
+            SendWeaponAnim(ACT_VM_PULLBACK);
+            WeaponSound(GetWeaponSound("charge"));
+        }
+    }
+    else
+    {
+        float flTotalChargeTime = gpGlobals->curtime - m_flChargeBeginTime;
+
+        if (flTotalChargeTime >= MOM_STICKYBOMB_MAX_CHARGE_TIME)
+        {
+            LaunchGrenade();
+        }
+    }
+}
+
+void CMomentumStickybombLauncher::SecondaryAttack()
+{
+    CMomentumPlayer *pPlayer = GetPlayerOwner();
+    if (m_iStickybombCount)
+    {
+        if (!pPlayer)
+            return;
+
+        if (DetonateRemoteStickybombs(false))
+        {
+            if (m_flLastDenySoundTime <= gpGlobals->curtime)
+            {
+                // Deny!
+                m_flLastDenySoundTime = gpGlobals->curtime + 1;
+                WeaponSound(GetWeaponSound("deny"));
+                return;
+            }
+        }
+        else
+        {
+            // Play a detonate sound.
+            WeaponSound(GetWeaponSound("detonate"));
+        }
+    }
+}
+
+float CMomentumStickybombLauncher::GetProjectileSpeed()
+{
+#ifdef CLIENT_DLL
+    static ConVarRef mom_sj_firing_mode("mom_sj_firing_mode");
+#endif
+
+    float flForwardSpeed = mom_sj_firing_mode.GetBool() ? RemapValClamped((gpGlobals->curtime - m_flChargeBeginTime), 
+                           0.0f, 
+                           MOM_STICKYBOMB_MAX_CHARGE_TIME, 
+                           MOM_STICKYBOMB_MIN_CHARGE_VEL, 
+                           MOM_STICKYBOMB_MAX_CHARGE_VEL) : 900.0f;
+
+    return flForwardSpeed;
+}
+
+
+//-----------------------------------------------------------------------------
+// Purpose: Add stickybombs to our list as they're fired
+//-----------------------------------------------------------------------------
+CMomStickybomb *CMomentumStickybombLauncher::FireProjectile(CMomentumPlayer *pPlayer)
+{
+    const auto pProjectile = FireStickybomb(pPlayer);
+#ifdef GAME_DLL
+    if (pProjectile)
+    {
+        // If we've gone over the max stickybomb count, fizzle the oldest
+        if (m_Stickybombs.Count() >= MOM_WEAPON_STICKYBOMB_COUNT)
+        {
+            CMomStickybomb *pTemp = m_Stickybombs[0];
+            if (pTemp)
+            {
+                pTemp->Fizzle();
+                pTemp->RemoveStickybomb(true);
+            }
+
+            m_Stickybombs.Remove(0);
+        }
+
+        CMomStickybomb *pStickybomb = (CMomStickybomb *)pProjectile;
+        pStickybomb->SetLauncher(this);
+
+        StickybombHandle hHandle;
+        hHandle = pStickybomb;
+        m_Stickybombs.AddToTail(hHandle);
+
+        m_iStickybombCount = m_Stickybombs.Count();
+    }
+#endif
+    return pProjectile;
+}
+
+CMomStickybomb *CMomentumStickybombLauncher::FireStickybomb(CMomentumPlayer *pPlayer)
+{
+#ifdef GAME_DLL
+    static ConVarRef cl_righthand("cl_righthand");
+#else
+    extern ConVar cl_righthand;
+#endif
```

Comment:
Does this ifdef need to happen if the only use of the `cl_righthand` convar is in the server? (GAME_DLL)